beets/dbcore/query.py

```diff
@@ -216,7 +217,7 @@ def value_match(cls, pattern: P, value: Any):
         """Determine whether the value matches the pattern. The value
         may have any type.
         """
-        return cls.string_match(pattern, util.as_string(value))
+        return cls.string_match(pattern, os.fsdecode(str(value or "")))
```

Comment:
I'm not sure about this, it definitely deviates from the previous behaviour:

- `str(b'abc') == "b'abc'"`
- Does it really make sense to involve `fsdecode` here? The `value` has no immediate relation to the filesystem, but is meant to match database fields (path queries are handled separately in `library.PathQuery`). Before, the `ignore` error handler was used: That way, there are definitely no decoding errors. Now, I'm not sure what could happen for decoding errors: Surrogates maybe? What would that mean for `string_match`.

Reply:
I just removed the last commit, keeping `as_string` in place :)